src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,70 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size = 0;
+    gchar *image_version = NULL;
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
+
+    if (xattr_size == -1)
```

Comment:
The second part is not second-guessing the `ssize_t` definition — it’s ensuring that the `xattr_size + 1` below doesn’t overflow.